src/test/java/com/commercetools/sync/producttypes/ProductTypeSyncOptionsBuilderTest.java

```diff
@@ -0,0 +1,243 @@
+import io.sphere.sdk.producttypes.ProductType;
+import io.sphere.sdk.producttypes.ProductTypeDraft;
+import io.sphere.sdk.producttypes.ProductTypeDraftBuilder;
+import io.sphere.sdk.producttypes.commands.updateactions.ChangeName;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ProductTypeSyncOptionsBuilderTest {
+    private static final SphereClient CTP_CLIENT = mock(SphereClient.class);
+    private ProductTypeSyncOptionsBuilder productTypeSyncOptionsBuilder = ProductTypeSyncOptionsBuilder.of(CTP_CLIENT);
+
+    @Test
+    public void of_WithClient_ShouldCreateProductTypeSyncOptionsBuilder() {
+        final ProductTypeSyncOptionsBuilder builder = ProductTypeSyncOptionsBuilder.of(CTP_CLIENT);
+        assertThat(builder).isNotNull();
+    }
+
+    @Test
+    public void build_WithClient_ShouldBuildProductSyncOptions() {
+        final ProductTypeSyncOptions productTypeSyncOptions = productTypeSyncOptionsBuilder.build();
+        assertThat(productTypeSyncOptions).isNotNull();
+        assertThat(productTypeSyncOptions.shouldAllowUuidKeys()).isFalse();
+        assertThat(productTypeSyncOptions.getBeforeUpdateCallback()).isNull();
+        assertThat(productTypeSyncOptions.getBeforeCreateCallback()).isNull();
+        assertThat(productTypeSyncOptions.getErrorCallBack()).isNull();
+        assertThat(productTypeSyncOptions.getWarningCallBack()).isNull();
+        assertThat(productTypeSyncOptions.getCtpClient()).isEqualTo(CTP_CLIENT);
+        assertThat(productTypeSyncOptions.getBatchSize()).isEqualTo(ProductTypeSyncOptionsBuilder.BATCH_SIZE_DEFAULT);
+    }
+
+    @Test
+    public void beforeUpdateCallback_WithFilterAsCallback_ShouldSetCallback() {
+        final TriFunction<List<UpdateAction<ProductType>>,
+                ProductTypeDraft, ProductType, List<UpdateAction<ProductType>>>
+            beforeUpdateCallback = (updateActions, newProductType, oldProductType) -> Collections.emptyList();
+        productTypeSyncOptionsBuilder.beforeUpdateCallback(beforeUpdateCallback);
+
+        final ProductTypeSyncOptions productTypeSyncOptions = productTypeSyncOptionsBuilder.build();
+        assertThat(productTypeSyncOptions.getBeforeUpdateCallback()).isNotNull();
+    }
+
+    @Test
+    public void beforeCreateCallback_WithFilterAsCallback_ShouldSetCallback() {
+        productTypeSyncOptionsBuilder.beforeCreateCallback((newProductType) -> null);
+
+        final ProductTypeSyncOptions productTypeSyncOptions = productTypeSyncOptionsBuilder.build();
+        assertThat(productTypeSyncOptions.getBeforeCreateCallback()).isNotNull();
+    }
+
+
+    @Test
+    public void allowUuid_WithFalse_ShouldSetFlag() {
```

Comment:
Do we need this test? You probably will need to remove it after this PR gets merged https://github.com/commercetools/commercetools-sync-java/pull/315

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/316#discussion_r223759942